### PR TITLE
FXIOS-3198: URI counts telemetry

### PR DESCRIPTION
**Pull request by nishant2718**
# Overview
This PR is in reference to https://github.com/mozilla-mobile/firefox-ios/issues/9158.

### CODE REVIEW

**Comment by nishant2718** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_

It will answer the number of URIs a user visits in one session. 

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_

This will help us understand how the app is used and how engaged users are. 

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_

There's no alternative, and no telemetry that can substitute here. 

_4) Can current instrumentation answer these questions?_

No.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>User has visited a URI</td>
    <td>Category 2</td>
    <td>#9158</td>
  </tr>
</table>

_6) How long will this data be collected?  Choose one of the following:_

Indefinitely. 

_7) What populations will you measure?_

All release channels and locales.

_8) If this data collection is default on, what is the opt-out mechanism for users?_

Users can opt of of data collection by disabling Usage and technical data from _Settings → Send Usage Data_.

_9) Please provide a general description of how you will analyze this data._

Glean 

_10) Where do you intend to share the results of your analysis?_

Only on Glean, mobile teams and BD have internal access.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_

No.

**Review by nbhasin2 — APPROVED** [review]

File: Client/metrics.yaml [hunk 1]
```diff
@@ -1267,6 +1267,17 @@ tabs:
...
+    type: counter
+    description: |
+      Record the number of URI's visited by the user.
+    bugs:
+      - https://github.com/mozilla-mobile/firefox-ios/issues/9158
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-ios/pull/9504
+    notification_emails:
+      - fx-ios-data-stewards@mozilla.com
+    expires: never
```
Comment (dnarcese) [inline Client/metrics.yaml:1280]: why are we not expiring this?

Comment (nishant2718) [reply Client/metrics.yaml:1280]: That's part of the acceptance criteria, but I can double check that.

Comment (nishant2718) [reply Client/metrics.yaml:1280]: Yup, it looks like NEVER is part of the acceptance criteria.

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Pull request merged**